raster/r.geomorphon/pattern.c

```diff
@@ -7,6 +7,73 @@
 static int nextr[NUM_DIRS] = { -1, -1, -1, 0, 1, 1, 1, 0 };
 static int nextc[NUM_DIRS] = { 1, 0, -1, -1, -1, 0, 1, 1 };
 
+/*
+ * In the two functions below return -1 if the nadir angle represents the
+ * terrain profile better than the zenith angle, or 1 in the opposite case,
+ * or 0 if neither is the case. Each angle must be non-negative.
+ */
+
+/*
+ * The simple comparison used in the original r.geomorphon implementation,
+ * although not in the original paper.
+ */
+static int scompare(const double nadir_angle, const double zenith_angle,
+                    const double nadir_threshold, const double zenith_threshold)
+{
+    if (zenith_angle > zenith_threshold || nadir_angle > nadir_threshold) {
+	if (nadir_angle < zenith_angle)
+	    return 1;
+	if (nadir_angle > zenith_angle)
+	    return -1;
+    }
+    return 0;
+}
+
+/*
+ * A more thorough comparison using several factors of different priority
+ * similar to the BGP best path selection algorithm.
+ */
+static int tcompare(const double na, const double za, const double nat,
```

Comment:
In scompare you are using full names but here shortening them. To improve readability of code, I would suggest to unify naming of semantically identical values in both functions.

Reply:
Point taken.